src/site/pages/digital/tests/containers/ExprToCircuitPopup.test.tsx

```diff
@@ -0,0 +1,143 @@
+import "@testing-library/jest-dom";
+import {act, render, screen}          from "@testing-library/react";
+import userEvent                      from "@testing-library/user-event";
+import {Provider}                     from "react-redux";
+import {applyMiddleware, createStore} from "redux";
+import thunk, {ThunkMiddleware}       from "redux-thunk";
+
+import {Setup} from "test/helpers/Setup";
+
+import {LED, ORGate, Switch} from "digital/models/ioobjects";
+
+import {OpenHeaderPopup} from "shared/state/Header";
+
+import {GetToggles} from "shared/tests/helpers/GetToggles";
+
+import {AppState} from "site/digital/state";
+
+import {AllActions} from "site/digital/state/actions";
+import {reducers}   from "site/digital/state/reducers";
+
+import {ExprToCircuitPopup} from "site/digital/containers/ExprToCircuitPopup";
+
+
+// beforeAll and beforeEach can be used to avoid duplicating store/render code, but is not recommended
+//  see: https://testing-library.com/docs/user-event/intro
+describe("Main Popup", () => {
+    const info = Setup();
+    const store = createStore(reducers, applyMiddleware(thunk as ThunkMiddleware<AppState, AllActions>));
+    const user = userEvent.setup();
+
+    beforeEach(() => {
+        render(<Provider store={store}><ExprToCircuitPopup mainInfo={info} /></Provider>);
+        act(() => {store.dispatch(OpenHeaderPopup("expr_to_circuit"))});
```

Comment:
unnecessary `{` and `}` since it's a singe-statement

Reply:
Removing them gives the following error:
```
Overload 1 of 2, '(callback: () => Promise<void>): Promise<undefined>', gave the following error.
  Type '{ readonly type: "OPEN_HEADER_POPUP_ID"; readonly popup: HeaderPopups; }' is missing the following properties from type 'Promise<void>': then, catch, finally, [Symbol.toStringTag]        
Overload 2 of 2, '(callback: () => VoidOrUndefinedOnly): void', gave the following error.
  Type '{ readonly type: "OPEN_HEADER_POPUP_ID"; readonly popup: HeaderPopups; }' is not assignable to type 'VoidOrUndefinedOnly'.
```

Reply:
Looks like an open issue https://github.com/microsoft/TypeScript/issues/12871, just add spaces around braces at least then